server.go

```diff
@@ -243,6 +245,36 @@ func (s *Server) register(sd *ServiceDesc, ss interface{}) {
 	s.m[sd.ServiceName] = srv
 }
 
+// ServiceMetadata returns the metadata for a service or method.
```

Comment:
we only have metadata for a service so that this is not accurate. method is mostly for sanity check. Once the caller inputs a method, this function would check if the input method is a method of the service.